homeassistant/components/tuya/binary_sensor.py

```diff
@@ -71,6 +75,34 @@ class TuyaBinarySensorEntityDescription(BinarySensorEntityDescription):
         ),
         TAMPER_BINARY_SENSOR,
     ),
+    # Dehumidifier
+    # https://developer.tuya.com/en/docs/iot/categorycs?id=Kaiuz1vcz4dha
+    "cs": (
+        TuyaBinarySensorEntityDescription(
+            key="tankfull",
+            dpcode=DPCode.FAULT,
+            device_class=BinarySensorDeviceClass.PROBLEM,
+            entity_category=EntityCategory.DIAGNOSTIC,
+            bitmap_key="tankfull",
+            translation_key="tankfull",
+        ),
```

Comment:
In the implementation by @MrAdam, an attempt was made to auto-detect all fault codes.
This leads to binary sensors being generated without translations